python/ray/data/read_api.py

```diff
@@ -2167,14 +2185,14 @@ def from_huggingface(dataset: "datasets.Dataset") -> MaterializedDataset:
         hf_ds_arrow = dataset.with_format("arrow")
         ray_ds = from_arrow(hf_ds_arrow[:])
```

Comment:
let's also add distributed reads for this case? create read tasks, and have each read task read a portion of `hf_ds_arrow`